File: Java/Easy/Accepted/number-of-even-and-odd-bits_8.java
class Solution {
    public int[] evenOddBit(int n) {
        
        int even = 0, odd = 0;
        int i = 0;
        while (n > 0) {
            if (i % 2 == 0 && n % 2 == 1) even++;
            if (i % 2 == 1 && n % 2 == 1) odd++;
            n = n / 2;
            i++;
        }
        return new int[]{even, odd};
    }
}

File: Rust/Easy/RuntimeError/number-of-even-and-odd-bits_8.rs
impl Solution {
    pub fn even_odd_bit(n: i32) -> Vec<i32> {
        
        let mut ret: Vec<i32> = vec![0; 2];
        let mut n = n;
        let mut i = 0;
        while n != 0 {
            if n & 1 == 1 {
                ret[i as usize] += 1;
            }
            n >>= 1;
            i += 1;
        }
        ret
    }
}